### PR TITLE
conf/machine: Add baremetal-riscv32nf config

### PR DESCRIPTION
**Pull request by agherzan**

### CODE REVIEW

**Comment by berolinux** [comment]
LGTM

File: conf/machine/baremetal-riscv32nf.conf [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+# qemuboot options
+QB_MACHINE = "-machine virt"
+QB_DEFAULT_KERNEL = "fw_payload.elf"
+QB_NETWORK_DEVICE = "-device virtio-net-device,netdev=net0,mac=@MAC@"
+QB_ROOTFS_OPT = "-drive file=@ROOTFS@,format=raw,id=hd0 -device virtio-blk-device,drive=hd0"
+QB_SLIRP_OPT = "-netdev user,id=net0,hostfwd=tcp::22222-:22"
+
+# Prelink does not yet work
+USER_CLASSES_remove = "image-prelink"
```
Comment (kraj) [inline conf/machine/baremetal-riscv32nf.conf:20]: this perhaps needs to be `:` for override char.

File: conf/machine/baremetal-riscv32nf.conf [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#@TYPE: Machine
+#@NAME: generic riscv32 baremetal machine without floating point instructions
+#@DESCRIPTION: Machine configuration for running a generic riscv32 without FP
+
+DEFAULTTUNE = "riscv32nf"
+require conf/machine/include/qemu.inc
+require conf/machine/include/riscv/tune-riscv.inc
+
+PREFERRED_VERSION_openocd-native = "riscv"
```
Comment (kraj) [inline conf/machine/baremetal-riscv32nf.conf:9]: override syntax needs to be corrected.

**Review by kraj — CHANGES_REQUESTED** [review]

File: conf/machine/baremetal-riscv32nf.conf (hunk 1 above)
Comment (agherzan) [reply conf/machine/baremetal-riscv32nf.conf:20]: Indeed. I'll fix.

File: conf/machine/baremetal-riscv32nf.conf (hunk 2 above)
Comment (agherzan) [reply conf/machine/baremetal-riscv32nf.conf:9]: Yes. I'll fix.

File: conf/machine/baremetal-riscv32nf.conf (hunk 1 above)
Comment (agherzan) [reply conf/machine/baremetal-riscv32nf.conf]: Fixed.

File: conf/machine/baremetal-riscv32nf.conf (hunk 2 above)
Comment (agherzan) [reply conf/machine/baremetal-riscv32nf.conf]: Fixed.

**Review by kraj — APPROVED** [review]

**Pull request merged**